test/optim/test_optim.py

```diff
@@ -333,6 +409,32 @@ def test_radam(self):
             ),
         )
 
+    def test_differentiable_lr(self):
+        params = torch.rand(10, requires_grad=True, dtype=torch.float64)
+        grad = torch.rand_like(params, requires_grad=True, dtype=torch.float64)
+
+        lr = torch.rand(1, requires_grad=True, dtype=torch.float64)
+        outer_mbuff = torch.rand_like(lr, requires_grad=True, dtype=torch.float64)
```

Comment:
That's a good point, there's no real reason to add in outer_state at all so I'll remove it

Reply:
I think we can definitely simplify the test by removing everything about testing the outer_optimizer! Interestingly though, I think it still has to be differentiable. I found that I think SGD only accepts non-leaf parameters (which I think anything that gets changed in the test function has to be because they're copied) if it's differentiable. I tried the following code to instantiate the optimizer in the debug console:

Code:
```py
print(p.is_leaf)  # False
SGD([p])
```
Output:
```
ValueError: can't optimize a non-leaf Tensor
```

Code:
```py
print(p.is_leaf)  # False
SGD([p], differentiable=True)
```
Output:
```
SGD (
Parameter Group 0
    dampening: 0
    differentiable: True
    foreach: None
    fused: None
    lr: 0.001
    maximize: False
    momentum: 0
    nesterov: False
    weight_decay: 0
)
```

Reply:
Outer_params should be leaves for the outer optimizer though, so that should not need to be differentiable!

Reply:
The reason I included this was that I think I have to clone lr so that the `outer_optimizer.step` at the end doesn't change the original lr, and the original clone makes lr a nonleaf.

When I don't clone lr it eventually ends up negative (which I don't really get why) and then the test fails because of a negative lr